invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/UserDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -37,23 +38,25 @@ export default class ItemsSearch extends Component {
   };
 
   executeSearch = queryString => {
+    queryString = queryString || this.props.queryString;
+    this.setState({ prevSearchQuery: queryString });
+
     if (queryString) {
       return this.fetchItems(queryString);
-    } else {
-      return this.fetchItems(this.props.queryString);
     }
   };
 
   _onPasteHandler = e => {
     let queryString = e.clipboardData.getData('Text');
     this.executeSearch(queryString).then(data => {
-      if (this.props.items.hits[0].status === 'LOANABLE') {
-        this.checkoutItem(this.props.items.hits[0], this.props.patron).then(
-          () => {
-            this.clearResults();
-            this.fetchUpdatedCurrentLoans(this.props.patron);
-          }
-        );
+      const { patron } = this.props;
```

Comment:
I think you could assign in outside the `then` or as class attribute if you don't like the long `this.props.patron`, because in this case it gets assigned every time on paste, but the patron does not change.